### PR TITLE
Added notebook to demonstrate sparse inference acceleration on 4th Gen Xeon

### PR DESCRIPTION
**Pull request by AlexKoff88**
- Notebook uses Optimum for OpenVINO to quantize the Neural Magic model
- The model is quantized using Optimum API (NNCF)
- Inference is run on 4 cores with 4 streams in parallel emulating AWS instance.

### CODE REVIEW

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (paularamo) [inline notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:1]: <p>Some general comments. </p><p> </p><ul><li>In between I have access to a 4th Gen Machine I will add some general comments here. </li><li>Could you please add a description cell here. Include links or figures to help to describe the further actions. </li><li>In functions could you please add docstring format and type hints, that will help readers to understand better the code without the need to debug the code.</li><li>Could you also add some comments in the code. Not all people are familiar with HuggingFace and Optimum.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/774/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (paularamo) [reply notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:1]: Thanks @AlexKoff88 for your contribution, I have started the process with OneCloud (48 hours), but DevCloud doesn't give me the option to access the machine without payment. So, I am figuring out how to access the 4th Gen. I think that also we need to include a note in the notebook to show the reader how they could test this notebook for free. The Tiny Machine option in DevCloud could be a broader solution now, but I found a gap in launching that instance. 

For now I added some comments to your notebook. As soon I have access I will test the notebook. Thanks and BR.

Comment (paularamo) [reply notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:1]: @AlexKoff88, trying to review your notebooks, I am facing some bottlenecks:

- Devcloud has a bug for free access, and it will be solved W3/4. (This will be the option for broader audiences)
- I contacted an internal team, and I could access that after Monday. 
- I am trying OneCloud, but I am facing some problems with the Git Proxy. And I am not sure if I could fix that on those machines.

I have added some comments on the notebook, but I think we need to add some notes about how users can access the processor.

I will jump into that once I have a solution. I am trying to be faster on this review, but I found a lot of bottlenecks.  :(

Best regards.

Comment (AlexKoff88) [reply notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:1]: Thanks @paularamo, I applied your comments related to the code. As for the device availability, I think this is a common problem and is not related directly to the 4th Gen Xeon, even though it is not widely spread. 
My assumption is that this tutorial is not for everybody for sure but it can be helpful for those who would like to get all the benefits of the new HW generation in case it is available.

**Review by paularamo — COMMENTED** [review]
I still have some issues accessing a 4th Gen machine. @AlexKoff88 could you please add the outputs so, that case, the devs can see the results? 

I think I will approve it, I don't want to create delays with this important topic.

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (eaidova) [inline notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #22.    # Create quantization config (default) and OVQuantizer
</pre><p>could you please provide some details about configuration and how it can be used (.e.g. which parameters can be additionally set?), for some users can be unclear why it is required create instance of class without any parameters and why it can not be done automatically inside pipeline)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/774/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: notebooks/116-sparsity-optimization/requirements.txt
Comment (eaidova) [inline notebooks/116-sparsity-optimization/requirements.txt:1]: does it uses nncf 2.3.0?
if yes, i think it can lead to some troubles (e.g. model onnx saving with torch 1.13 not supported) and break some notebooks whiich requires new api...

Please also consider to install requirements inside notebook

Comment (AlexKoff88) [reply notebooks/116-sparsity-optimization/requirements.txt:1]: I am waiting for the upcoming NNCF release.

**Comment by AlexKoff88** [comment]
> 

I've added the output. You can check the speedup.

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (AlexKoff88) [reply notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:1]: I've added extra info about the config. It is not possible to explain all the possible configurations here. The default should work OOB and, btw, I can remove it at all so that it will be created inside the quantize method.

File: notebooks/116-sparsity-optimization/requirements.txt
Comment (AlexKoff88) [reply notebooks/116-sparsity-optimization/requirements.txt]: Moved to notebook.

**Review by paularamo — APPROVED** [review]
Thanks @AlexKoff88

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:11]:
```suggestion
    "# Accelerate Inference of Sparse Transformer Models with OpenVINO™ and 4th Gen Intel&reg; Xeon&reg; Scalable Processors\n",
```
Just to have consistent formatting in the titles..

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:12]:
```suggestion
    "This tutorial demonstrates how to improve performance of sparse Transformer models with [OpenVINO](https://docs.openvino.ai/) on 4th Gen Intel® Xeon® Scalable processors. It uses a pre-trained model from the [Hugging Face Transformers](https://huggingface.co/transformers/) library and shows how to convert it to the OpenVINO™ IR format and run inference on a CPU using a dedicated runtime option that enables sparsity optimizations. It also demonstrates how to get more performance stacking sparsity with 8-bit quantization. To simplify the user experience, the [Hugging Face Optimum](https://huggingface.co/docs/optimum) library is used to convert the model to OpenVINO™ IR format and quantize it using [Neural Network Compression Framework](https://github.com/openvinotoolkit/nncf). It consists of the following steps:\n",
```

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:15]:
```suggestion
    "- Download and quantize sparse BERT model from a public source using the OpenVINO integration with Hugging Face Optimum.\n",
```

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:163]:
```suggestion
    "## Quantize the model using Hugging Face Optimum API\n",
```

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:164]:
```suggestion
    "The sparsity acceleration MatMul operations are only available cases when these operations are quantized into 8-bit precision. If the model is not in 8-bit precision, it can be quantized using either method available for OpenVINO models. For more details, please refer to the [Model Optimization Guide](https://docs.openvino.ai/latest/openvino_docs_model_optimization_guide.html). In this tutorial we use the Hugging Face Optimum API to quantize the model. The Hugging Face Optimum API is a high-level API that allows us to convert and quantize models from the Hugging Face Transformers library to the OpenVINO™ IR format. For more details refer to the [Hugging Face Optimum documentation](https://huggingface.co/docs/optimum/intel/optimization_ov)."
```

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:244]:
```suggestion
    "Benchmark dense inference performance using parallel execution on 4 CPU cores to simulate a small cloud-based instance. Sequence length is set to 16 which is common for multiple use cases, e.g. conversational AI."
```

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:478]:
```suggestion
    "This feature can improve inference performance for models with sparse weights in scenarios where the model is deployed to handle multiple requests in parallel asynchronously. It is especially helpful in the case of small sequence length, e.g. 32 and lower.\n",
```

File: notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/116-sparsity-optimization.ipynb:480]:
```suggestion
    "For more details about asynchronous inference with OpenVINO please refer to the following documentation:\n",
```

File: notebooks/116-sparsity-optimization/README.md
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/README.md:1]:
```suggestion
# Accelerate Inference of Sparse Transformer Models with OpenVINO™ and 4th Gen Intel&reg; Xeon&reg; Scalable Processors
```

File: notebooks/116-sparsity-optimization/README.md
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/README.md:3]:
```suggestion
This tutorial demonstrates how to improve performance of sparse Transformer models with [OpenVINO](https://docs.openvino.ai/) on 4th Gen Intel® Xeon® Scalable processors. It uses a pre-trained model from the [Hugging Face Transformers](https://huggingface.co/transformers/) library and shows how to convert it to the OpenVINO™ IR format and run inference on a CPU using a dedicated runtime option that enables sparsity optimizations. It also demonstrates how to get more performance stacking sparsity with 8-bit quantization. To simplify the user experience, the [Hugging Face Optimum](https://huggingface.co/docs/optimum) library is used to convert the model to the OpenVINO™ IR format and quantize it using [Neural Network Compression Framework](https://github.com/openvinotoolkit/nncf).
```

File: notebooks/116-sparsity-optimization/README.md
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/README.md:5]:
```suggestion
>**NOTE**: This tutorial requires OpenVINO 2022.3 or newer and 4th Gen Intel&reg; Xeon&reg; Scalable processor that can be acquired on [Amazon Web Services (AWS)](https://aws.amazon.com/ec2/instance-types/r7iz/).
```

File: notebooks/116-sparsity-optimization/README.md
Comment (ryanloney) [inline notebooks/116-sparsity-optimization/README.md:11]:
```suggestion
- Download and quantize sparse BERT model from the public using OpenVINO integration with Hugging Face Optimum.
```

**Review by ryanloney — CHANGES_REQUESTED** [review]
A few minor edits

**Review by ryanloney — APPROVED** [review]

**Comment by AlexKoff88** [comment]
I don't know what is required on top of what has already been done here. CI is red but as far as I can see it should be green after the merge. @eaidova, @paularamo please take a look.

**Comment by eaidova** [comment]
@AlexKoff88 please update one more readme file: https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/README.md

**Review by eaidova — APPROVED** [review]

File: notebooks/README.md
Comment (eaidova) [inline notebooks/README.md:96]:
```suggestion
| [116-sparsity-optimization](116-sparsity-optimization/)| Improve performance of sparse Transformer models |
```

**Pull request merged**
